src/coroutine.js

```diff
@@ -1,25 +1,21 @@
+import Action from './Action'
+
 export default function (resolve, iterator, promise) {
 	new Coroutine(resolve, iterator, promise).run()
+	// taskQueue.add(new Coroutine(resolve, iterator, promise))
 	return promise
 }
 
-class Coroutine {
+class Coroutine extends Action {
 	constructor (resolve, iterator, promise) {
+		super(promise)
 		this.resolve = resolve
-		this.iterator = iterator
-		this.promise = promise
+		this.next = iterator.next.bind(iterator)
+		this.throw = iterator.throw.bind(iterator)
```

Comment:
Ooops, I had reverted to `this.generator` exactly because of the detrimal perf effect both [here](https://github.com/bergus/creed/commit/e61b095eaf0ed278d07b2d1bebef6746997768e2) and [there](https://github.com/bergus/creed/commit/4b6c9573962d6599ea27e2c7c19138c85c86a6d3) in my branches and completely forgot about this. I guess I should simply revert this part and wipe it from history through rebaseing.
I would like to avoid `.call` for standard callbacks as there have been issues with host objects that did not implement it ([example](http://stackoverflow.com/q/5472938/1048572), guess what browser), and `call.call(f, context)` doesn't look clean/fast either.